### PR TITLE
Finish fixing graph/classic refresh inconsistencies

### PR DESCRIPTION
**Pull request by agrare**
Set the vm storage_id and the snapshot parent link.

### CODE REVIEW

File: app/models/manageiq/providers/microsoft/inventory/collector/infra_manager.rb [hunk 1]
```diff
@@ -59,6 +59,23 @@ def hosts
     hosts_by_host_name.values
   end
 
+  def storage_id_by_host_name_and_mount_point
+    @storage_id_by_host_name_and_mount_point ||= begin
+      drive_letter = /\A[a-z][:]/i
+
```
Comment (djberg96) [inline app/models/manageiq/providers/microsoft/inventory/collector/infra_manager.rb]: I can't remember, is a UNC path a possibility here?

Comment (agrare) [reply app/models/manageiq/providers/microsoft/inventory/collector/infra_manager.rb]: Possibly?  I lifted this from the existing logic.  For now I'm just looking for exact equality between the refresh strategies.

Comment (djberg96) [reply app/models/manageiq/providers/microsoft/inventory/collector/infra_manager.rb]: Ok, if it hasn't come up before then don't worry about it for now. :)

File: spec/models/manageiq/providers/microsoft/infra_manager/refresher_spec.rb [hunk 2]
```diff
@@ -50,14 +50,14 @@
   end
 
   it "Classic same as graph refresh" do
-    stub_settings_merge(:ems_refresh => {:scvmm => {:inventory_object_refresh => false}})
-    EmsRefresh.refresh(@ems)
-    inventory_after_classic_refresh = serialize_inventory
-
```
Comment (agrare) [inline spec/models/manageiq/providers/microsoft/infra_manager/refresher_spec.rb]: Running classic refresh first was covering up some cases where graph refresh wasn't setting some attribute and it was passing as a false positive

**Pull request merged**